contracts/core/MarginRequirements.sol

```diff
@@ -0,0 +1,223 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity =0.6.10;
+
+pragma experimental ABIEncoderV2;
+
+import {Ownable} from "../packages/oz/Ownable.sol";
+import {OtokenInterface} from "../interfaces/OtokenInterface.sol";
+import {OracleInterface} from "../interfaces/OracleInterface.sol";
+import {AddressBookInterface} from "../interfaces/AddressBookInterface.sol";
+import {MarginVault} from "../libs/MarginVault.sol";
+import {SafeMath} from "../packages/oz/SafeMath.sol";
+import {ERC20Interface} from "../interfaces/ERC20Interface.sol";
+
+/**
+ * @title MarginRequirements
+ * @author Ribbon Team
+ * @notice Contract that defines margin requirements and operations
+ */
+contract MarginRequirements is Ownable {
+    using MarginVault for MarginVault.Vault;
+    using SafeMath for uint256;
+
+    OracleInterface public oracle;
+
+    /************************************************
+     *  STORAGE
+     ***********************************************/
+
+    /// @notice AddressBook module
+    address public addressBook;
+
+    ///@dev mapping between a hash of (underlying asset, collateral asset, isPut) and a mapping of an account to an initial margin value
+    mapping(bytes32 => mapping(address => uint256)) public initialMargin;
+    ///@dev mapping between an account owner and a mapping of a specific vault id to a maintenance margin value
+    mapping(address => mapping(uint256 => uint256)) public maintenanceMargin;
```

Comment:
let us actually change maintenanceMargin to `mapping(uint256 => uint256) public maintenanceMargin;`

If we do it by vaultID, since the owner of every vault is OTC_WRAPPER, there won't be any problems by just referenced vaultID

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/be9ff20f8188f6835647c10266bb69672a9cfe5c